### DISCUSSION TITLE
Printing multiple files not working

### DISCUSSION THREAD

**Discussion by mcp292** [OpenPrinting/cups#634] [Q&A]
# Problem
I cannot print multiple files `lp(r) file1.pdf file2.pdf`. I can print single files `lp(r) file1.pdf`. 

When printing multiple files, the printer will sometimes audibly "gear up" and then stop. Observing `lpstat -p`, I see "Waiting for job to complete" when sent and then "Print job canceled at printer." shortly after. Concurrently, `lpq` shows a job and then "no entries".

I've also noted that about a minute after these cancelled jobs, sometimes the printer audibly gears up as if it's feeding paper, and maintains a whir for about 15 seconds, then goes to sleep.

# Setup
I have a Brother HL-L2370DW. 

I have set up `avahi` by following [these instructions](https://wiki.archlinux.org/title/Avahi#Hostname_resolution) (making sure to disable and stop `systemd-resolve.service`). I connected over the network using the [`dnssd` URI](http://localhost:631/help/network.html) and the `everywhere` driver.

# Tried
1. Upgrading and restarting system.
2. Removing printer (`lpadmin -x`).
3. Resetting printer (factory, network, machine).
4. Re-adding printer.

# Notes
There is a [firmware update available](https://support.brother.com/g/b/producttop.aspx?c=us&lang=en&prod=hll2370dw_us), but I don't have a way to update it on Linux as far as I'm aware (they only provide tools for Mac and Windows and I only have access to Linux machines).

**Comment by zdohnal** [comment]
Hi,

what CUPS version do you have?
What OS?
Would you mind enabling CUPS debug logging with ```cupsctl LogLevel=debug2``` and trying to print the multiple files again?

Thank you in advance!

**Comment by mcp292** [comment]
Certainly!

CUPS 2.4.2  
ArchLinux

This is some of the most recent output of `/var/log/cups/error_log` after issuing `cupsctl LogLevel=debug2`, and `lp file1.txt file2.txt`:


EDIT: Please use attaching feature, otherwise the thread is unreadable.

[error_log_short.txt](https://github.com/OpenPrinting/cups/files/10955651/error_log_short.txt)

**Comment by mcp292** [comment]
This is the entire file with `LogLevel=debug`:

EDIT: Copied long log is unreadable in a thread, use file attachments.

[error_log_full.txt](https://github.com/OpenPrinting/cups/files/10955673/error_log_full.txt)

**Comment by zdohnal** [comment]
Hi,

the error_log says the job was canceled on the printer due document-format-error - it seems some machines are affected by it (I tested it on HP Color LaserJet Pro MFP M277dw and it worked for me).

Japhier opened an issue for it, let's merge it there - https://github.com/OpenPrinting/cups/issues/643 .
